final-project/scripts/main.js

```diff
@@ -1,10 +1,110 @@
-import "../styles/main.scss";
-
-// testing that `const` and arrow functions transpile correctly
-const before = "Hello -world!";
-const after = before
-  .split("")
-  .filter(char => char !== "-")
-  .join("");
-console.log({ before });
-console.log({ after });
+import "../styles/main.scss";
+import Axios from "axios";
+
+function showPage() {
+  document.getElementById("body-layout").classList.toggle("layout-wrapper--loading");
+  document.getElementById("page-loader").style.display = "none";
+}
+
+window.onload = function () {
+  //Wait just to simulate a low connection and a delay
+  setTimeout(showPage, 3000);
+
+  //Retrieve blog posts
+  const url = 'https://wt-4662f45b9eefda7172b747b28d23efdb-0.sandbox.auth0-extend.com/blog';
+  Axios
+    .get(url)
+    .then(data => {
+      const articles = data.data.articles;
+      for (let i = 0; i < articles.length; i++) {
+        const article = articles[i];
+
+        // Add articles
+        $('#main-blog-articles')
+          .append(`
+            <li id="blog-article-${i+2}" class="main-blog-carousel-slides__slide">

```

Comment:
Probably it would have been more readable and maintainable to have these divs cloned and transformed from the HTML instead of creating them from scratch with JS, just in case you need to change their structure or classes.